pkg/steps/new_e2e.go

```diff
@@ -66,14 +75,20 @@ func (s *newE2ETestStep) Run(ctx context.Context, dry bool) error {
 		return nil
 	}
 	var errs []error
-	if err := s.runSteps(ctx, s.pre); err != nil {
+	var err error
+	if err = s.runSteps(ctx, s.pre, ""); err != nil {
 		errs = append(errs, fmt.Errorf("%q pre steps failed: %v", s.name, err))
-	} else {
-		if err := s.runSteps(ctx, s.tests); err != nil {
+	}
+	var secret string
+	if secret, err = createSecret(s.secretClient, s.jobSpec.Namespace, s.artifactDir); err != nil {
+		errs = append(errs, fmt.Errorf("failed to create secret: %v", err))
+	}
+	if errs == nil {
```

Comment:
The logic here is getting a bit nuts. Why not just return when we can't make the secret?

Reply:
Depends if we want `post` if `pre` failed.